server/src/main/java/com/thoughtworks/go/server/domain/PluginSettings.java

```diff
@@ -101,13 +77,16 @@ public void addConfigurations(PluginInfo pluginInfo, List<ConfigurationProperty>
         if (pluginInfo != null) {
             ConfigurationPropertyBuilder builder = new ConfigurationPropertyBuilder();
             for (ConfigurationProperty property : configurationProperties) {
-                settingsMap.add(builder.create(property.getConfigKeyName(), property.getValue(), null, false));
+                final PluginConfiguration configuration = configPropertyFor(property.getConfigKeyName(), pluginInfo);
+                if (configuration != null) {
+                    settingsMap.add(builder.create(property.getConfigKeyName(), property.getConfigValue(), property.getEncryptedValue(), configuration.isSecure()));
+                }
```

Comment:
We are ignoring a property which does not have a configuration in pluginInfo. Instead we ignoring probably we should leave it to plugin to validate it.